merge-v1/src/mapping.ts

```diff
@@ -1,74 +1,268 @@
-  entity.alphaMass = event.params.alphaMass;
-
-  // Entities can be written to the store with `.save()`
-  entity.save();
-
-  // Note: If a handler doesn't require existing field values, it is faster
-  // _not_ to load the entity from the store. Instead, create it fresh with
-  // `new Entity(...)`, set the fields that should be updated and save the
-  // entity back to the store. Fields that were not set or unset remain
-  // unchanged, allowing for partial updates to be applied.
-
-  // It is also possible to access smart contracts from mappings. For
-  // example, the contract that has emitted the event can be connected to
-  // with:
-  //
-  // let contract = Contract.bind(event.address)
-  //
-  // The following functions can then be called on this contract to access
-  // state variables and other data:
-  //
-  // None
+  return mergeClass;
 }
 
-export function handleApproval(event: Approval): void {}
+// /**
+//  * @notice object type where properties represent value-based-decoded nft metadata
+//  * TODO: not sure but I am getting a compile error here. Asked DK. I need to just look into this. This will be done in next PR as I clean up the code.
+//  */
+// type Values = {
+//   massSize: BigInt;
+//   mergeClass: string;
+//   color: string;
+// };
 
-export function handleApprovalForAll(event: ApprovalForAll): void {}
+// /**
+//  * @notice carries out decoding of value for a particular mass NFT
+//  * @returns value : Values
+//  * @dev TODO: see Object Type Values above and the error that is there. Once that is resolved then this can be used.
+//  */
+// export function decodeValue(nftValue: BigInt): Values {
+//   let value: Values;
+//   let _mergeClass: BigInt = nftValue / CLASS_MULTIPLIER;
+//   let mergeClass = checkMergeClass(_mergeClass);
 
-export function handleConsecutiveTransfer(event: ConsecutiveTransfer): void {}
+//   value = {
+//     massSize: (nftValue % CLASS_MULTIPLIER) as BigInt,
+//     mergeClass: mergeClass,
+//     color: checkColor(mergeClass),
+//   };
+
+//   return value;
+// }
 
-export function handleMassUpdate(event: MassUpdate): void {}
+/* ========== EVENT HANDLERS ========== */
 
+/**
+ * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
+ * @param event: Transfer
+ */
 export function handleTransfer(event: Transfer): void {
-  // when _safeMint() is called in base contract erc721, it emits transfer event where from is address(0). Try using this to load new NFTs and Users entities
+  let _to = event.params.to;
+  let _from = event.params.from;
+
+  let to = _to.toHex();
+  let from = _from.toHex();
+  let nft: NFT;
+
+  let user = User.load(to);
+  let tokenId = event.params.tokenId.toString();
+
+  // checks if this is a new user
+  if (!user) {
+    user = new User(to);
+    nft = getNFT(event, user);
+  } // checks if this is a mint() scenario
+  else if (from == EMPTY_ADDRESS) {
+    nft = createNFT(event, user);
+  }
```

Comment:
Good points. Yes, this can be cleaned up due to the redundancy of the current implementation. I need to think about this more and how to go about it. Below is the context of which the current if/else statements handle.

1. `if (!user)` handles:
  - when Nifty Gateway mints a mergeNFT for the first time. Ultimately an NFT is created.
  - when Nifty Gateway is transferring an NFT to an external HD wallet account.
  - when someone new buys a mergeNFT from OpenSea or whatever, or is transferred to a new HD wallet account. 
  - when address(0) is used for the first time in a burn scenario (transfer to address(0)).
2. second `if` handles:
  - All minting times after the first time that Nifty Gateway mints.
3. third `if` handles:
  - All other scenarios incl: when one user transfers an NFT to a pre-existing mergeNFT holder.
  - Sending to address(0) after the first burn ever done within this collection.
  - Sending to address(dead) temporarily